pkg/clients/iam/iam_client.go

```diff
@@ -148,3 +152,93 @@ func (client *IAMClient) CreateBucketAccess(ctx context.Context, userName, bucke
 
 	return accessKeyOutput, nil
 }
+
+// RevokeBucketAccess is a helper that revokes bucket access by orchestrating individual steps to delete the user, inline policy, and access keys.
+func (client *IAMClient) RevokeBucketAccess(ctx context.Context, userName, bucketName string) error {
+	err := client.EnsureUserExists(ctx, userName)
+	if err != nil {
+		return err
+	}
+
+	err = client.DeleteInlinePolicy(ctx, userName, bucketName)
+	if err != nil {
+		return err
+	}
+
+	err = client.DeleteAllAccessKeys(ctx, userName)
+	if err != nil {
+		return err
+	}
+
+	err = client.DeleteUser(ctx, userName)
+	if err != nil {
+		return err
+	}
+
+	klog.InfoS("Successfully revoked bucket access", "user", userName, "bucket", bucketName)
+	return nil
+}
+
+func (client *IAMClient) EnsureUserExists(ctx context.Context, userName string) error {
+	_, err := client.IAMService.GetUser(ctx, &iam.GetUserInput{UserName: &userName})
+	if err != nil {
+		var noSuchEntityErr *types.NoSuchEntityException
+		if errors.As(err, &noSuchEntityErr) {
+			klog.InfoS("IAM user does not exist, nothing to revoke", "user", userName)
+			return nil
+		}
```

Comment:
I am gonna remove this conditon as it is not required, we should send back the error
Working on it.

Reply:
Fixed this